ibis/expr/api.py

```diff
@@ -3274,8 +3274,53 @@ def _struct_get_field(expr, field_name):
     return ops.StructField(expr, field_name).to_expr().name(field_name)
 
 
-def _destructure(expr: StructColumn) -> DestructColumn:
-    """ Destructure a ``Struct`` to create a destruct column.
+def _struct_get_attr(expr: StructValue, attr_name: str) -> ValueExpr:
+    """Get field named `attr_name` from the ``StructValue`` expression `expr`.
+
+    Parameters
```

Comment:
also *why* are you defining ```__dir__``` on StructValueExpr in the first place? this is the cause of the trouble. it would be *much* better to provide an attribute which holds the data itself, e.g. ```.attr``` or something, then you don't have these namespace collisions.

Reply:
Do you mean why we define `StructValue.__getattr__`? (and that we should **remove** `StructValue.__getattr__`, and only allow access to fields of the struct through `StructValue.__getitem__`?)

The underlying struct is already stored in the way you describe if I'm understanding correctly. Information about the underlying struct is stored in an attribute called `StructValue._dtype`, and then doing `my_struct_value.foo` or `my_struct_value['foo']` constructs a operation+expression (using information from `StructValue._dtype`) that would pull the field's value out when executed.

Getting rid of `StructValue.__getattr__` sounds like a good solution to me. It would simplify all of this (no need for any pickling-specific fix). I would have to do a little digging to see whether it's used anywhere.

Reply:
no the current implementation is a hack.

Reply:
> Getting rid of StructValue.__getattr__ sounds like a good solution to me. It would simplify all of this (no need for any pickling-specific fix). I would have to do a little digging to see whether it's used anywhere.

yes do this, its *way* better